### PR TITLE
Support TransformOperations animations in Brush animator

### PR DESCRIPTION
**Pull request by maxkatz6**
Previously that PR added support for Brush.Transform - https://github.com/AvaloniaUI/Avalonia/pull/6344, but animations support wasn't added as it's not really possible to interpolate matrixes. Meanwhile, we already have interpolation algorithm for the limited set of TransformOperations.

Saying that, with this PR it's not possible to animate brushes with TransformOperations (`Transform="rotate(180deg)"`), but not TransformGroup.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027932-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Animation/Animators/GradientBrushAnimator.cs [hunk 1]
```diff
@@ -72,6 +73,25 @@ public override IDisposable BindAnimation(Animatable control, IObservable<IGradi
...
+        {
+            if (oldTransform is TransformOperations oldTransformOperations
+                && newTransform is TransformOperations newTransformOperations)
+            {
+
+                return new ImmutableTransform(TransformOperations
+                    .Interpolate(oldTransformOperations, newTransformOperations, progress).Value);
+            }
+
+            if (oldTransform is { })
```
Comment (robloo) [inline src/Avalonia.Base/Animation/Animators/GradientBrushAnimator.cs:87]: This is a nit so feel free to ignore.

However, I wonder why are you all still using this syntax? It's widely not liked and is very unintuitive. Its usage only made some sense for a small period of time between language versions. With the addition of the "not" keyword (C# 9.0) it also really shouldn't be done anymore and most projects don't.

Besides, I think `!(oldTransform is null)` was always the most accepted syntax or more rarely `!object.ReferenceEquals(object, null)`. These are more readable alternatives.

Comment (maxkatz6) [reply src/Avalonia.Base/Animation/Animators/GradientBrushAnimator.cs:87]: I guess it's just a habit. I would prefer "is not null" here, yes.

Comment (maxkatz6) [inline src/Avalonia.Base/Animation/Animators/GradientBrushAnimator.cs:87]:
```suggestion
            if (oldTransform is not null)
```

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027941-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028032-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029093-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**